client/data/marketplace/use-site-search-es-query.ts

```diff
@@ -0,0 +1,217 @@
+import phpUnserialize from 'phpunserialize';
+import { UseQueryResult, UseQueryOptions, InfiniteData, useInfiniteQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import { extractSearchInformation, normalizePluginsList } from 'calypso/lib/plugins/utils';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import {
+	DEFAULT_FIRST_PAGE,
+	DEFAULT_PAGE_SIZE,
+	DEFAULT_CATEGORY,
+	BASE_STALE_TIME,
+	SITE_SEARCH_CACHE_KEY,
+	SITE_SEARCH_ENDPOINT,
+} from './constants';
+import { ESHits, ESResponse, Plugin, PluginQueryOptions } from './types';
+import { getPluginsListKey } from './utils';
+
+/**
+ *
+ * @param pluginSlug
+ * @param iconsArray
+ * @returns An object containing a resolution -> icon url map
+ */
+const createIconsObject = ( pluginSlug: string, iconsArray: string ) => {
+	type Icon = {
+		resolution: string;
+		filename: string;
+		location: string;
+		revision: string;
+	};
+	const icons: Record< string, Icon > = phpUnserialize( iconsArray );
+	return Object.values( icons ).reduce(
+		( prev: Record< string, string >, { resolution, filename, location, revision } ) => {
+			prev[
+				resolution
+			] = `https://ps.w.org/${ pluginSlug }/${ location }/${ filename }?rev=${ revision }`;
+			return prev;
+		},
+		{}
+	);
+};
+
+const createAuthorUrl = ( header_author: string, header_author_uri: string ) =>
+	`<a href="${ header_author_uri }">${ header_author }</a>`;
+
+const mapIndexResultsToPluginData = ( results: ESHits ): Plugin[] => {
+	return results.map( ( { fields: hit } ) => {
+		return {
+			name: hit.title_en, // TODO: add localization
+			slug: hit.slug,
+			version: hit.stable_tag,
+			author: createAuthorUrl(
+				hit[ 'meta.header_author.value' ],
+				hit[ 'meta.header_author_uri.value' ]
+			),
```

Comment:
Do we have a `contributors` field? 

We saw we were getting better results when using it to find the author for search.

Reference: https://github.com/Automattic/wp-calypso/pull/62777#issuecomment-1109980813

Reply:
I've tried using the function, or implementing it again for the ES index. The index has something similar, the problem is that the data is in a different structure:

There's `taxonomy.plugin_contributors` and a flat `contributors` array:
```json
{
    "taxonomy": {
        "plugin_contributors": [
            {
                "name": "automattic",
                "slug": "automattic",
                "term_id": "77494"
            },
            {
                "name": "mikejolley",
                "slug": "mikejolley",
                "term_id": "77602"
            }
        ]
    },
    "contributors": [
        "automattic",
        "mikejolley"
    ]
}
```

Also, the `meta.header_author` object contains a display name, while the rest of the fields contain the slug/username. In the case of Automattic it's just an upper/lower case difference, but there are authors for which these two are entirely different. Without a slug field to match on, it's hard to choose the correct contributor as the author from the list.
```json
"header_author": {
    "value": [
        "Automattic"
    ],
    "long": [
        0
    ],
    "double": [
        0
    ],
    "boolean": [
        true
    ]
},
```

I'll look into how the contributors array is ordered, maybe the header author is always the first contributor.